### ISSUE TITLE
Enhance /v1/node rest apis for clusters that are setup to autoscale

### ISSUE THREAD

**Issue by vgankidi** [prestosql/presto#521]
We have clusters that autoscale based on workload. Some of the problems we face:
a) Workers that are gracefully shutdown are also considered as "Failed" by the failure detector. We have dashboards to track failed nodes to help with debugging issues. This metric is unreliable now since it includes deactivated nodes by administrators as well.
b) Also v1/node endpoint has a ton of workers that have already been deactivated/decommissioned. Our auto scale service hits these /v1/node endpoints to make scaling decisions. It would be nice to know directly from the coordinator which workers are active/failed/decommissioned.
c) Have a way to bring back nodes in shutting_down to active when we detect an increased workload.

We have a patch internally that does the following:
a) Decommissioned nodes do not appear in /v1/node/failed endpoint
b) Expose /v1/node/decommissioned endpoint
c) Expose /v1/node/active endpoint

I see that DiscoveryNodeManager depends on the failure detector and all it does is to get the current list of nodes in different states used for scheduling and planning. Node states are polled every 5 seconds in DiscoveryNodeManager. HeartbeatFailureDetector runs every 500ms and pings remote nodes. The uri that it pings seems to be http://IP:PORT which actually returns a 404. 

In our internal patch, we get the modified the ping uri to http://IP:PORT/v1/info/state. This allows us to have enough information in the failure detector to enhance the /node/failed endpoint and add the /node/active and /node/decommissioned endpoints. I'm happy to upload our internal patch or discuss other approaches to tackle these issues.

Does it make sense to combine the DiscoveryNodeManager and HeartbeatFailureDetector into one?

**Comment by vgankidi** [comment]
cc @martint @electrum

**Comment by martint** [comment]
@vgandiki, I’ll respond later today with some thoughts.

**Comment by martint** [comment]
I don't think we want to merge DiscoveryNodeManager with the the HeartbeatFailureDetector, but we could certainly improve how they interact and what they expose. Currently, the failure detector is very primitive (based on just a regular ping), but we want it enhance it to take into account other signals such as whether a given node is causing a large portion of query failures or is an outlier across a variety of other metrics. 

We should figure out how we want to signal that nodes are active/inactive, but we should certainly filter the `/v1/node/failed` view to exclude any node that's not active.

**Comment by tooptoop4** [comment]
care to share the patch @vgankidi ?
